Modules/socketmodule.c

```diff
@@ -1639,7 +1639,6 @@ idna_converter(PyObject *obj, struct maybe_idna *data)
         return 1;
     }
     data->obj = NULL;
-    len = -1;
```

Comment:
This is dead code now, but `len` gets set in a bunch of different conditional branches. I am mildly worried that something might change in the future where someone forgets to set `len` in one of those branches and we end up using uninitialized memory.

Doesn't look *terribly* dangerous, but maybe it would make sense to leave it as `len = -1` and then add `assert(len >= 0)` directly before `len` is used?